File: go/226. Invert Binary Tree.go
package main

func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return root
	}
	tmp := root.Left
	root.Left = root.Right
	root.Right = tmp
	if root.Left != nil {
		invertTree(root.Left)
	}
	if root.Right != nil {
        invertTree(root.Right)
	}
	return root
}

File: js/226. Invert Binary Tree.js
function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}

/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function (root) {
    if (root === null) {
        return root;
    }
    let tmp = root.left;
    root.left = root.right;
    root.right = tmp;
    if (root.left !== null) {
        invertTree(root.left);
    }
    if (root.right !== null) {
        invertTree(root.right);
    }
    return root;
};